iis-wf/iis-wf-referenceextraction/src/main/java/eu/dnetlib/iis/wf/referenceextraction/eoscservices/ServiceDBBuilder.java

```diff
@@ -0,0 +1,46 @@
+package eu.dnetlib.iis.wf.referenceextraction.eoscservices;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Map;
+
+import org.apache.commons.lang.StringUtils;
+
+import com.google.common.base.Preconditions;
+
+import eu.dnetlib.iis.importer.schemas.Service;
+import eu.dnetlib.iis.wf.referenceextraction.AbstractDBBuilder;
+
+/**
+ * Process creating a database reading {@link Service} input avro records.
+ *
+ * @author mhorst
+ */
+public class ServiceDBBuilder extends AbstractDBBuilder<Service> {
+    
+    private final static String PARAM_SCRIPT_LOCATION = "scriptLocation";
+
+    // -------------------------- CONSTRUCTORS ------------------------------
+
+    public ServiceDBBuilder() {
+        super(Service.SCHEMA$, "service", "service_db");
+    }
+
+    // -------------------------- LOGIC -------------------------------------
+
+    @Override
+    public ProcessExecutionContext initializeProcess(Map<String, String> parameters) throws IOException {
+        String scriptLocation = parameters.get(PARAM_SCRIPT_LOCATION);
+        Preconditions.checkArgument(StringUtils.isNotBlank(scriptLocation),
+                "sql script location not provided, '%s' parameter is missing!", PARAM_SCRIPT_LOCATION);
+
+        String targetDbLocation = System.getProperty("java.io.tmpdir") + File.separatorChar + "services.db";
+        File targetDbFile = new File(targetDbLocation);
+        targetDbFile.setWritable(true);
```

Comment:
This method does not work for nonexistent files, so is the file supposed to exist already?

Reply:
As far as I vaguely remember it was required at some point so it looks like under some circumstances (idk, maybe running this process on the same datanode which run this process before?) this file might already exist.

This piece of code also originates from other `*DBBuilder` class so what I could do is to extract this to a dedicated common method and create a dedicated issue to take a closer look at this.

Reply:
Done: https://github.com/openaire/iis/issues/1399

I don't want to spin another batch of refactoring as a part of #1398 so I am delegating this task to become part of this newly created issue.